### DISCUSSION TITLE
How to get UV coordinates of pixels with MeshRenderer?

### DISCUSSION THREAD

**Discussion by xhuan8** [facebookresearch/pytorch3d#1011] [General]
Hi,

I want to get UV coordinates of pixels instead of colors.
Is there any existing configurations for this feature? 

`Imports
from pytorch3d.renderer import (
    FoVPerspectiveCameras, look_at_view_transform,
    RasterizationSettings, BlendParams,
    MeshRenderer, MeshRasterizer, HardPhongShader
)

Initialize an OpenGL perspective camera.
R, T = look_at_view_transform(2.7, 10, 20)
cameras = FoVPerspectiveCameras(device=device, R=R, T=T)

Define the settings for rasterization and shading. Here we set the output image to be of size
512x512. As we are rendering images for visualization purposes only we will set faces_per_pixel=1
and blur_radius=0.0. Refer to rasterize_meshes.py for explanations of these parameters.
raster_settings = RasterizationSettings(
    image_size=512,
    blur_radius=0.0,
    faces_per_pixel=1,
)
renderer = MeshRenderer(
    rasterizer=MeshRasterizer(cameras=cameras, raster_settings=raster_settings),
    shader=HardPhongShader(device=device, cameras=cameras)
)`

**Comment by bottler** [comment]
I think https://github.com/facebookresearch/pytorch3d/issues/629 shows what you are looking for.

**Reply by xhuan8** [reply]
Thanks, it works. 
with above code just replace lighting with AmbientLights, to avoid UV coordinates changed by lighting .
